src/app/admin/components/system-configuration/event-mappings/add-update-event-mapping/add-update-event-mapping.component.ts

```diff
@@ -0,0 +1,169 @@
+import { EventMappingsService } from '@services/event-mappings/event-mappings.service';
+import { FormService } from '@services/form/form.service';
+import { HeaderService } from '@services/header/header.service';
+import { optionalMaxLengthValidator } from '@validators/optional-maxlength.validator';
+import { Subscription, combineLatest, map } from 'rxjs';
+
+@Component({
+  selector: 'app-add-update-event-mapping',
+  standalone: true,
+  templateUrl: './add-update-event-mapping.component.html',
+  styleUrl: './add-update-event-mapping.component.scss',
+  imports: [ReactiveFormsModule, GovukHeadingComponent, CommonModule, ValidationErrorSummaryComponent],
+})
+export class AddUpdateEventMappingComponent implements OnInit, OnDestroy {
+  fb = inject(FormBuilder);
+  router = inject(Router);
+  headerService = inject(HeaderService);
+  eventMappingsService = inject(EventMappingsService);
+  formService = inject(FormService);
+  errorMessageService = inject(ErrorMessageService);
+  destroyRef = inject(DestroyRef);
+
+  private mappingTypes: Partial<EventMapping[]> = [];
+
+  eventMappingsPath = 'admin/system-configuration/event-mappings';
+
+  eventHandlers$ = this.eventMappingsService.getEventHandlers();
+  eventMappings$ = this.eventMappingsService
+    .getEventMappings()
+    .pipe(map((eventMappings) => this.assignTypes(eventMappings)));
+
+  error$ = this.errorMessageService.errorMessage$;
+
+  data$ = combineLatest({
+    eventHandlers: this.eventHandlers$,
+    eventMappings: this.eventMappings$,
+  });
+
+  isSubmitted = false;
+  isRevision = false;
+  context: 'create' | 'edit' = 'create';
+
+  errorSubscription: Subscription | null = null;
+
+  form = this.fb.nonNullable.group({
+    type: ['', [Validators.required, optionalMaxLengthValidator(255)]],
+    subType: ['', [optionalMaxLengthValidator(255)]],
+    eventName: ['', [Validators.required, optionalMaxLengthValidator(255)]],
+    eventHandler: ['', [Validators.required]],
+    withRestrictions: [false, [Validators.required]],
+  });
+
+  onSubmit() {
+    this.errorMessageService.clearErrorMessage();
+    this.form.markAllAsTouched();
+
+    if (this.form.valid) {
+      this.eventMappingsService.createEventMapping(this.form.value as EventMappingFormValues).subscribe(() => {
+        this.router.navigate([this.eventMappingsPath], { queryParams: { newEventMapping: true } });
+      });
+    }
+
+    if (this.isSubmitted) return; // prevent multiple subscriptions
+
+    this.getErrorSummary();
+    this.isSubmitted = true;
+  }
+
+  onCancel() {
+    this.router.navigate([this.eventMappingsPath]);
+  }
+
+  ngOnInit(): void {
+    this.headerService.hideNavigation();
+    this.setUniqueTypeValidation();
+
+    this.errorSubscription = this.error$.subscribe((error) => {
+      this.setResponseErrors(error);
+    });
+  }
+
+  ngOnDestroy(): void {
+    if (this.errorSubscription) {
+      this.errorSubscription.unsubscribe();
```

Comment:
Instead of doing this you can do:
`.pipe(takeUntilDestroyed(this.destroyRef))`
save having to manually call unsubscribe() in ngOnDestroy() lifecycle hook